### ISSUE TITLE
Split layout with percentage height

### ISSUE THREAD

**Issue by SlartThunder** [ibhagwan/fzf-lua#1040]
### Info

- Operating System: MacOS
- Shell: omz
- Terminal: iTerm
- `nvim --version`:NVIM v0.9.4
- `fzf --version`:0.42.0 (d471067)

- [x] The issue is reproducible with `mini.sh`

### Description 
I recently switched from NVIM from FZF to use FZF-LUA. 

In FZF you can set the layout to be a floating window, or stick to a position in current buffer.  
E.g: let g:fzf_layout = { 'down': '50%'}
The above command would open FZF from down the active buffer, which is very handy when you have splits all around and you want to change the buffer in that particular window.

**Comment by ibhagwan** [comment]
You can use `winopts.split` to pass any command (including percentages), for example:
```lua
require'fzf-lua'.setup({
  -- "belowright new"  : split below
  -- "aboveleft new"   : split above
  -- "belowright vnew" : split right
  -- "aboveleft vnew   : split left
  winopts = { split = 'botright new' }
})
```

**Comment by SlartThunder** [comment]
Thanks, I tried with 
split  = "belowright new | horizontal resize 30%"
but when having already a split, being active in the upper one, for example, when opening FZFLua it "floats" the buffer size, rather than using the 30% of the current one…
I hope I make myself understand…

**Comment by ibhagwan** [comment]
> Thanks, I tried with split = "belowright new | horizontal resize 30%" but when having already a split, being active in the upper one, for example, when opening FZFLua it "floats" the buffer size, rather than using the 30% of the current one… I hope I make myself understand…

Resize doesn’t support percentage, if you want 30% you have to use a fraction of `lines`.
 
 If that’s what you want it’s better done with lua code, I can craft the 2 line function for you when I’m not AFK.

**Comment by ibhagwan** [comment]
Here you go, modify this to your liking:
```lua
require("fzf-lua").setup({
  winopts_fn = function()
    -- local split = "botright new" -- use for split under **all** windows
    local split = "belowright new"  -- use for split under current window
    local height = math.floor(vim.o.lines * 0.3)
    return { split = split .. " | resize " .. tostring(height) }
  end,
})
```

**Comment by SlartThunder** [comment]
Thanks for your code, though it didn't work as I expected.
Nevertheless, it gave me a hint to implement what I'm trying to achieve.
Basically, I'd like to have a different FZF setup based on status of NVIM, if it has splits or not.

I have a FZFLua custom config I call with
`
local preview_v = "nohidden"
local fullscreen_v = false

local function reflection_function()
  return { preview = { hidden = preview_v}, fullscreen = fullscreen_v }
end

local config = function()
  require'fzf-lua'.setup {
  -- fzf_bin         = 'sk',            -- use skim instead of fzf?
                                        -- https://github.com/lotabout/skim
                                        -- can also be set to 'fzf-tmux'
  winopts_fn = reflection_function,  
 ...
 ...
`

Then triggering FZF with the map (map('n', 'ø', ':lua require\'fzf\'.fzf_setup_base_on_splits()<cr>', options)):

`
local function is_buffer_in_split()
    local current_win = vim.api.nvim_get_current_win()
    local current_buf = vim.api.nvim_win_get_buf(current_win)
    local current_win_width = vim.api.nvim_win_get_width(current_win)
    local total_width = vim.api.nvim_get_option("columns")

    return current_win_width ~= total_width
end

M.fzf_setup_base_on_splits = function()
  if is_buffer_in_split() then
    --reflection("hidden", true)
    preview_v = "hidden"
    fullscreen_v = true
  else
    --reflection("nohidden", false)
    preview_v = "nohidden"
    fullscreen_v = false
  end
  
  config()
  require('fzf-lua').buffers()
end
`
So far so good, it changes the config of FZF when there are split windows.
Though, my intention is to have no preview and fullscreen in the active buffer split.

I am not very skilled with Lua and NVIM, so may it be better options to achieve the same…

**Comment by ibhagwan** [comment]
> So far so good, it changes the config of FZF when there are split windows.
Though, my intention is to have no preview and fullscreen in the active buffer split.

Your code seems to do just that, but you need to call `setup_base_on_splits` before calling any fzf function, why not call it inside the `refelecrion_funcfion`?
```lua
local function reflection_function()
  local preview_v
  local fullscreen_v
  if is_buffer_in_split() then
    preview_v = "hidden"
    fullscreen_v = true
  else
    preview_v = "nohidden"
    fullscreen_v = false
  end
  return { preview = { hidden = preview_v}, fullscreen = fullscreen_v }
end
```

This way you can use this with any fzf-lua function without extra code

> I am not very skilled with Lua and NVIM, so may it be better options to achieve the same…

Lua is very simple, anyone can get the hang of it pretty quickly.

**Comment by SlartThunder** [comment]
Nice improvement on the comment above, indeed!!
But FZFLua opens in NEOVIM's fullscreen rather than in buffers fullscreen. Is there a way to achieve such thing?

**Comment by ibhagwan** [comment]
> Nice improvement on the comment above, indeed!! But FZFLua opens in NEOVIM's fullscreen rather than in active window buffer's fullscreen. Is there a way to achieve such thing?

Unfortunately no such option, `winopts.fullscreen` is always neovim fullscreen, filling up a certain buffer would require more involved code.

**Comment by ibhagwan** [comment]
> I am not sure whether I understand… But the point would be to use FZF buffers the same netwr does, opening itself in current active split to allow selection of another opened buffer. Not to spawn to an additional window

Yes, that’s exactly what this would enable but you’d probably need to write more code for it than you’re currently prepared.

**Comment by johslarsen** [comment]
@SlartThunder I stumbled upon a similar problem, and ended up using fzf-tmux for splitting instead of the builtin one, and this works nicely with a configuration like e.g.:
```lua
require 'fzf-lua'.setup {
  'fzf-tmux',
  fzf_tmux_opts = { ["-d"] = "30%" },
  fzf_opts = { -- just to style it more like how fzf-tmux from the terminal looks like
    ["--border"] = "none",
    ["--color"] = "dark",
    ["--layout"] = "default",
  },
}
```
Note that there seems to be an fzf bug that is fixed upstream (worked in 0.46.1.r22.gf35a9da9-1, but not with the current version in the archlinux repository: 0.46.1-1) where the default fzf-tmux layout (split below 50%) works, but if any of the `-u/d/l/r` arguments are specified it falls back to a non-tmux fzf view, which does not work well with `fzf-lua`

**Comment by ibhagwan** [comment]
> @SlartThunder I stumbled upon a similar problem, and ended up using fzf-tmux for splitting instead of the builtin one

Great idea @johslarsen, the only difference is that fzf-tmux can only use “native” fzf previewers so you lose the ability to have the Sam’s color scheme in the previewer.
